### PR TITLE
First pass at function to make pretty Fig 3 (and figure for supplement)

### PR DESCRIPTION
**Pull request by cstillwellusgs**
Not finished yet but I do need your feedback. (1) I'm not sure I calculated the error bars quite the same way as you, but they looks similar. (2) General confusion about the raw metrics (full-year) and where the seasons come into play. Let me know if you want to chat this afternoon or later this week and screenshare to make sure I'm computing everything correctly.

### CODE REVIEW

**Comment by cstillwellusgs** [comment]
cluster_map_barplot_seasonal

**Comment by jds485** [comment]
I just did a quick check on the quantiles used and it looks like you did the same method as I used (5th and 95th quantiles computed over all gages in the region). For the full-year metrics, seasons are only used for plotting purposes. I think that figure would be a supplement instead of a main figure

File: 3_cluster/src/seasonal_metric_cluster.R [hunk 1]
```diff
@@ -725,3 +725,188 @@ get_colmeans_panel_plt <- function(metric_names, metric_mat, by_quantile,
...
+      summarise(average = mean(mean), 
+                lower = mean(low_bound), 
+                upper = mean(high_bound), 
+                .groups = "drop")
+    metric_summary$metric_name <- factor(metric_summary$metric_name, 
+                                         levels = c("Duration", "Frequency", 
+                                                    "Total Volume", "Max. Flow"))
+    metric_summary$season <- factor(metric_summary$season, 
+                                    levels = c("OND", "JFM", "AMJ", "JAS"))
+    barplot <- ggplot(data = metric_summary) +
```
Comment (jds485) [inline 3_cluster/src/seasonal_metric_cluster.R:863]: Let's fix the ylim to be 0, 1

File: 3_cluster.R [hunk 2]
```diff
@@ -1215,5 +1215,33 @@ p3_targets_list <- list(
...
+                              seasonal = TRUE,
+                              season_months = season_months,
+                              num_clusters = 5,
+                              dir_out = '3_cluster/out/manuscript_plots'),
+             deployment = 'main',
+             format = 'file'
+  ),
+  #full year metrics (not seasonal; for supplement)
+  tar_target(p3_fullyear_cluster_map_barplot_quants_agg_selected_png,
+             plot_map_barplot(gages = p1_feature_vars_g2,
```
Comment (jds485) [inline 3_cluster.R:1235]: Not sure this edit is right
```suggestion
             plot_map_barplot(gages = p1_feature_vars_g2_sf,
```

File: 3_cluster/src/seasonal_metric_cluster.R [hunk 3]
```diff
@@ -725,3 +725,188 @@ get_colmeans_panel_plt <- function(metric_names, metric_mat, by_quantile,
...
 }
 
+plot_map_barplot <- function(gages, cluster_table, metric_mat, metric_names, 
+                             metric_quants_agg, seasonal, season_months, 
+                             num_clusters, dir_out) {
+  
+  #' @description creates .png image containing map of cluster regions and their
+  #' respective barplots (across season, if applicable) for the metrics
+  #' 
+  #' @param gages dataframe of the gages (rows)
```
Comment (jds485) [inline 3_cluster/src/seasonal_metric_cluster.R:735]: as sf?

File: 3_cluster.R [hunk 4]
```diff
@@ -1215,5 +1215,33 @@ p3_targets_list <- list(
...
+                              cluster_table = p3_gages_clusters_quants_agg_raw_metrics,
+                              metric_mat = p2_FDC_metrics,
```
Comment (jds485) [inline 3_cluster.R:1237]: I think this needs to be seasonal metrics to compute the seasonal barplot. Maybe this is where your confusion was?
```suggestion
                              metric_mat = p2_FDC_metrics_season, 
```

**Review by jds485 — CHANGES_REQUESTED** [review]
I have some initial comments that we can discuss in the meeting

**Comment by cstillwellusgs** [comment]
cluster_map_barplot_seasonal

**Comment by cstillwellusgs** [comment]
cluster_map_barplot_fullyear

**Review by jds485 — APPROVED** [review]
Looks good!

**Pull request merged**